src/Trip/Utility.ts

```diff
@@ -0,0 +1,13 @@
+import { Day } from '.';
+import { addDaysToDate } from '../Util';
+
+export function decorateDaysWithDate(tripStartDate: string | null, days: Day[] | null): Day[] | null {
+	if (!tripStartDate || !days) {
```

Comment:
`!tripStartDate` shouldn't be here, it would cause to remove days from trip without dates. And also you are using it later in code.